src/main/java/dinkplugin/notifiers/TradeNotifier.java

```diff
@@ -0,0 +1,160 @@
+    }
+
+    public void reset() {
+        image.lazySet(null);
+    }
+
+    public void onTradeMessage(String message) {
+        if (!TRADE_ACCEPTED_MESSAGE.equals(message) || !isEnabled()) {
+            this.reset();
+            return;
+        }
+
+        ItemContainer tradeInv = client.getItemContainer(InventoryID.TRADE);
+        ItemContainer otherInv = client.getItemContainer(InventoryID.TRADEOTHER);
+        if (tradeInv == null && otherInv == null) {
+            log.debug("Could not find traded items!");
+            this.reset();
+            return;
+        }
+        Item[] trade = tradeInv != null ? tradeInv.getItems() : new Item[0];
+        Item[] other = otherInv != null ? otherInv.getItems() : new Item[0];
+        long receiveValue = getTotalValue(other);
+        long giveValue = getTotalValue(trade);
+        long grossValue = receiveValue + giveValue;
+        long netValue = receiveValue - giveValue;
+        if (grossValue < config.tradeMinValue()) {
+            this.reset();
+            return;
+        }
+        List<SerializedItemStack> received = getItems(other);
+        List<SerializedItemStack> discarded = getItems(trade);
+
+        String localPlayer = client.getLocalPlayer().getName();
+        String counterparty = client.getVarcStrValue(TRADE_COUNTERPARTY_VAR);
+        Template content = Template.builder()
+            .template(config.tradeNotifyMessage())
+            .replacementBoundary("%")
+            .replacement("%USERNAME%", Replacements.ofText(localPlayer))
+            .replacement("%COUNTERPARTY%", Replacements.ofLink(counterparty, config.playerLookupService().getPlayerUrl(counterparty)))
+            .replacement("%GROSS_VALUE%", Replacements.ofText(QuantityFormatter.quantityToStackSize(grossValue)))
+            .replacement("%NET_VALUE%", Replacements.ofText(QuantityFormatter.quantityToStackSize(netValue)))
+            .build();
+
+        createMessage(config.tradeSendImage(), NotificationBody.builder()
+            .text(content)
+            .extra(new TradeNotificationData(counterparty, received, discarded, grossValue, netValue))
+            .playerName(localPlayer)
+            .screenshotOverride(image.get())
+            .type(NotificationType.TRADE)
+            .build()
+        );
+
+        this.reset();
+    }
+
+    public void onWidgetLoad(WidgetLoaded event) {
+        if (event.getGroupId() == TRADE_CONFIRMATION_GROUP) {
+            boolean chatHidden = Utils.hideWidget(config.screenshotHideChat(), client, ComponentID.CHATBOX_FRAME);
+            boolean whispersHidden = Utils.hideWidget(config.screenshotHideChat(), client, PRIVATE_CHAT_WIDGET);
+            drawManager.requestNextFrameListener(frame -> {
+                image.set(frame);
+                Utils.unhideWidget(chatHidden, client, clientThread, ComponentID.CHATBOX_FRAME);
+                Utils.unhideWidget(whispersHidden, client, clientThread, PRIVATE_CHAT_WIDGET);
+            });
```

Comment:
We should make sure to test this with trade delays disabled, where both parties accept the trade as fast as possible

Reply:
That edge case would probably require playing at very low fps to realistically reproduce

Reply:
works fine _without_ very low FPS from what I can test